diff --git a/stagit-index.c b/stagit-index.c
index e019793..0e2522e 100644
--- a/stagit-index.c
+++ b/stagit-index.c
@@ -71,6 +71,9 @@ writeheader(FILE *fp)
 	fputs("<!DOCTYPE html>\n"
 		"<html>\n<head>\n"
 		  "<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\" />\n"
+		  "<meta name=\"referrer\" content=\"none\" />\n"
+		   "<meta name=\"viewport\" content=\"width=device-width\" />\n"
+
 		"<title>", fp);
 	xmlencode(fp, description, strlen(description));
 	fprintf(fp, "</title>\n<link rel=\"icon\" type=\"image/png\" href=\"%sfavicon.png\" />\n", relpath);
diff --git a/stagit.c b/stagit.c
index 86c3920..aced01c 100644
--- a/stagit.c
+++ b/stagit.c
@@ -327,6 +327,8 @@ writeheader(FILE *fp, const char *title)
 	 fputs("<!DOCTYPE html>\n"
 		   "<html>\n<head>\n"
 		   "<meta http-equiv=\"Content-Type\" content=\"text/html; charset=UTF-8\" />\n"
+		   "<meta name=\"referrer\" content=\"none\" />\n"
+		   "<meta name=\"viewport\" content=\"width=device-width\" />\n"
 		   "<title>", fp);
 	 xmlencode(fp, title, strlen(title));
 	 if (title[0] && strippedname[0])
@@ -1018,7 +1020,7 @@ main(int argc, char *argv[])
 	 const git_error *e = NULL;
 	 mode_t mask;
 	 FILE *fp, *fpread;
-	char path[PATH_MAX], repodirabs[PATH_MAX + 1], *p;
+	 char path[PATH_MAX], repodirabs[PATH_MAX + 1], *p, *pwd;
 	 char tmppath[64] = "cache.XXXXXXXXXXXX", buf[BUFSIZ];
 	 size_t n;
 	 int i, fd;
@@ -1043,6 +1045,11 @@ main(int argc, char *argv[])
 			   if (errno == ERANGE && (nlogcommits == LLONG_MAX ||
 									   nlogcommits == LLONG_MIN))
 					usage(argv[0]);
+		  } else if (argv[i][1] == 'C') {
+			   if (i+1 < argc)
+					pwd = argv[++i];
+			   else
+					errx(EXIT_FAILURE, "no path specified");
 		  }
 	 }
 	 if (!repodir)
@@ -1127,6 +1134,9 @@ main(int argc, char *argv[])
 					  git_object_type(obj) == GIT_OBJ_BLOB);
 	 git_object_free(obj);
 
+	 if (chdir(pwd))
+		  err(EXIT_FAILURE, "couldn't change to %s", pwd);
+
 	 /* log for HEAD */
 	 fp = efopen("log.html", "w");
 	 relpath = "";
